File: SpringBootBackend/src/main/java/com/th/entity/Vendor.java
//package com.th.entity;
//
//import jakarta.persistence.Column;
//import jakarta.persistence.Entity;
//import jakarta.persistence.EnumType;
//import jakarta.persistence.Enumerated;
//import lombok.AllArgsConstructor;
//import lombok.Data;
//import lombok.Getter;
//import lombok.NoArgsConstructor;
//import lombok.Setter;
//
//@Data
//@AllArgsConstructor
//@NoArgsConstructor
//@Getter
//@Setter
//@Entity
//public class Vendor extends BaseEntity {
//    private String companyName;
//
//    @Column(unique = true, nullable = false)
//    private String email;
//
//    private String password;
//
//    private String phone;
//
//    @Column(unique = true, nullable = false)
//    private String gstin;
//
//    private String companyOwnerName;
//
//    @Column(unique = true, nullable = false)
//    private String ownerAadharNumber;
//
//    @Column(unique = true, nullable = false)
//    private String panNumber;
//
//    private String city;
//
//    private String amount;
//
//    @Enumerated(EnumType.STRING)
//    private Status status = Status.PENDING;
//    
//    
//    public enum Status {
//        PENDING, APPROVED, SUSPENDED
//    }
//}

package com.th.entity;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class Vendor extends BaseEntity {

    @NotEmpty(message = "Company name is required")
    private String companyName;

    @Email(message = "Email should be valid")
    @NotEmpty(message = "Email is required")
    @Column(unique = true, nullable = false)
    private String email;

    //@Pattern(regexp = "^[a-zA-Z]\\w{3,14}$", message = "Password should be atlleast 4 characters , and no longer than 15 charaters.")
    @NotEmpty(message = "Password is required")
    private String password;

    @Pattern(regexp = "^[0-9]{10}$", message = "Phone number must be 10 digits")
    private String phone;

    @Pattern(regexp = "^[0-9]{2}[A-Z]{5}[0-9]{4}[A-Z]{1}[1-9A-Z]{1}[Z]{1}[0-9A-Z]{1}$", message = "Invalid GSTIN format")
    @NotEmpty(message = "GSTIN is required")
    @Column(unique = true, nullable = false)
    private String gstin;

    @NotEmpty(message = "Company owner name is required")
    private String companyOwnerName;

    @Pattern(regexp = "^[0-9]{12}$", message = "Aadhar number must be 12 digits")
    @NotEmpty(message = "Owner Aadhar number is required")
    @Column(unique = true, nullable = false)
    private String ownerAadharNumber;

    @Pattern(regexp = "^[A-Z]{5}[0-9]{4}[A-Z]{1}$", message = "Invalid PAN number format")
    @NotEmpty(message = "PAN number is required")
    @Column(unique = true, nullable = false)
    private String panNumber;

    @NotEmpty(message = "City is required")
    private String city;

    
    private BigDecimal amount;
    
    private String comment;
    
    private String resetToken; // Stores password reset token
    
    private boolean isEmailVerified = false; // New field
    
    private String resetTokenExpiry; // Expiry time for

    @Enumerated(EnumType.STRING)
    private Status status = Status.PENDING;

    public enum Status {
        PENDING, APPROVED, SUSPENDED , REJECTED
    }
}

File: DOTNET BACKEND/TransitHive/Models/Vendor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;
using TransitHive.Interfaces;

namespace TransitHive.Models
{
    [Index(nameof(Gstin), IsUnique = true)]
    [Index(nameof(OwnerAadharNumber), IsUnique = true)]
    [Index(nameof(PanNumber), IsUnique = true)]
    [Index(nameof(Email), IsUnique = true)]
    public class Vendor : IAccount
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Company Name is required")]
        [StringLength(100, ErrorMessage = "Company name cannot be longer than 100 characters")]
        [RegularExpression("^[a-zA-Z0-9 ]*$", ErrorMessage = "Company name can only contain letters, numbers, and spaces")]
        public string CompanyName { get; set; }

        [Required(ErrorMessage = "Email is required")]
        [EmailAddress(ErrorMessage = "Email should be valid")]
        public string Email { get; set; }

        [Required(ErrorMessage = "Password is required")]
        [RegularExpression("^(?=.*[A-Za-z])[A-Za-z\\d@$!%*?&]{6,}$", ErrorMessage = "Password must be at least 6 characters long and contain at least one alphabet")]
        public string Password { get; set; }

        [Required(ErrorMessage = "Phone is required")]
        [RegularExpression("^[6-9][0-9]{9}$", ErrorMessage = "Phone number must be 10 digits and start with 6, 7, 8, or 9")]
        public string Phone { get; set; }

        [Required(ErrorMessage = "GSTIN is required")]
        [RegularExpression("^[0-9]{2}[A-Z]{5}[0-9]{4}[A-Z]{1}[1-9A-Z]{1}[Z]{1}[0-9A-Z]{1}$", ErrorMessage = "Invalid GSTIN format")]
        public string Gstin { get; set; }
        public string CompanyOwnerName { get; set; }

        [Required(ErrorMessage = "Owner Aadhar number is required")]
        [RegularExpression("^[0-9]{12}$", ErrorMessage = "Aadhar number must be 12 digits")]
        public string OwnerAadharNumber { get; set; }

        [Required(ErrorMessage = "PAN number is required")]
        [RegularExpression("^[A-Z]{5}[0-9]{4}[A-Z]{1}$", ErrorMessage = "Invalid PAN number format")]
        public string PanNumber { get; set; }

        [Required(ErrorMessage = "State is required")]
        public string State { get; set; }

        [Required(ErrorMessage = "City is required")]
        [StringLength(100, ErrorMessage = "City name cannot be longer than 100 characters")]
        [RegularExpression("^[a-zA-Z ]*$", ErrorMessage = "City name can only contain letters and spaces")]
        public string City { get; set; }

        [Required(ErrorMessage = "Address is required")]
        public string Address { get; set; }

        public string Amount { get; set; }
        public string? Comments { get; set; }

        public bool IsEmailVerified { get; set; } = false; // New field
        public string ResetToken { get; set; } // Stores password reset token
        public DateTime? ResetTokenExpiry { get; set; } // Expiry time for token

        [Column(TypeName = "ENUM('PENDING', 'APPROVED', 'SUSPENDED', 'REJECTED')")]
        public Status Status { get; set; } = Status.PENDING;
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public ICollection<Booking> Bookings { get; set; }
        public ICollection<Payment> Payments { get; set; }
        public ICollection<Review> Reviews { get; set; }
    }

    public enum Status { PENDING, APPROVED, SUSPENDED, REJECTED }
}
